src/Raven.Server/Documents/Handlers/Processors/Revisions/AbstractRevisionsHandlerProcessorForGetRevisionsBin.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Diagnostics;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Exceptions.Documents.Revisions;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Revisions
+{
+    internal abstract class AbstractRevisionsHandlerProcessorForGetRevisionsBin<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler, TOperationContext>
+        where TRequestHandler : RequestHandler
+        where TOperationContext : JsonOperationContext
+    {
+        public AbstractRevisionsHandlerProcessorForGetRevisionsBin([NotNull] TRequestHandler requestHandler, [NotNull] JsonContextPoolBase<TOperationContext> contextPool) : base(requestHandler, contextPool)
+        {
+        }
+
+        protected abstract ValueTask GetAndWriteRevisionsBinAsync(TOperationContext context, long etag, int pageSize);
+
+        protected abstract bool IsRevisionsConfigured();
+
+        protected abstract void AddPagingPerformanceHint(PagingOperationType operation, string action, string details, long numberOfResults, int pageSize, long duration,
+            long totalDocumentsSizeInBytes);
+
+        public override async ValueTask ExecuteAsync()
+        {
+            if (IsRevisionsConfigured() == false)
+                throw new RevisionsDisabledException();
+
+            var start = RequestHandler.GetLongQueryString("etag", required: false) ?? 0;
```

Comment:
very confusing, I think we should throw if someone sends 'etag', this is now `start`, right?

Reply:
I wanted to leave it at `etag` so it won't break back comp but I'll change it to `start` and add it to this issue:
https://issues.hibernatingrhinos.com/issue/RavenDB-18514

Do you mean throw a "deprecated" error to the user?

Reply:
Changed to GetStart()

Reply:
@lastav5 let's throw if we get the `etag` parameter

Reply:
Done